Cluster registry changes and tests

aeweston-msft opened the pull request:
## Changes introduced by this PR
- Added wrapper functions to call the ClusterRegDeleteValue, ClusterRegCreateBatch, ClusterRegCloseBatch and ClusterRegBatchAddCommand Cluster API functions from go
- Added exported functions which expose the new functionality, as well as a LoadValues function which exposes the functionality of ClusterRegEnumValue

## Test Plan
- Added tests to check the behaviour of the LoadValues, DeleteValue, CreateBatch, AddBatchCommand and CloseBatch methods that I added
- Ran the new tests successfully with modifications to use a resource of my local cluster rather than validResourceName (did not include this change in PR)

Commit:
Cluster registry changes and tests

KnicKnic reviewed (changes requested):
Overall looks good. Thanks for the PR!

I requested a few minor changes, and please fix the unsafe usage.

aeweston-msft commented on pkg/cluster/registry.go:
```diff
@@ -152,6 +165,22 @@ func clusterRegEnumValue(handle KeyHandle, index uint32) (keyName string, dwType
 }
 
+// LoadValues loads the values and data of a key into a map
+func (handle KeyHandle) LoadValues() (map[string][]byte, error) {
```
Could you please clarify what you would like to see changed? As far as I can see, the set value methods all take byte array's as input (except SetGuidValue). Do you mean that dwType from clusterRegEnumValue should also be returned?

KnicKnic replied:
yes I mean dwType should also be returned.

aeweston-msft replied:
OK, thanks for the clarification!

Commit:
changes for review comments

aeweston-msft commented:
Thanks for the review comments! I have addressed them to the best of my knowledge and updated the PR. Please let me know if you would like me to make any further changes. Otherwise I can squash the two commits together so that it can be merged.

KnicKnic commented:
Thanks for the changes!
